pkg/kubelet/cm/event_dispatcher_test.go

```diff
@@ -0,0 +1,488 @@
+					testCase.replyError, eventReply.Error)
+			}
+		}
+	}
+}
+
+func TestEventDispatcher_PostStopPod(t *testing.T) {
+	server, err := runDummyServer()
+	if err != nil {
+		t.Skipf("Cannot run GRPC server: %q", err)
+	}
+	defer server.Close()
+
+	isolator := &lifecycle.RegisterRequest{
+		Name:          "isolator1",
+		SocketAddress: server.GetAddress(),
+	}
+
+	ed := newEventDispatcher()
+	ed.isolators = make(map[string]*registeredIsolator)
+	ed.Register(ctx.Background(), isolator)
+
+	testCases := []struct {
+		serverError error
+	}{
+		{
+			serverError: nil,
+		},
+		{
+			serverError: fmt.Errorf("Server Error"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		server.ResetDummyObject()
+		server.SetServerError(testCase.serverError)
+
+		err := ed.PostStopPod("/")
+		if err != nil {
+			if !strings.Contains(err.Error(), testCase.serverError.Error()) {
+				t.Errorf("Expected error(%q) isn't message of grpc error(%q)",
+					testCase.serverError,
+					err)
+			}
+		}
+		if server.GetLastEventFromServer() == nil {
+			t.Error("Lifecycle object should be accessable in lifecycle handler, but it wasn't")
+		}
+
+	}
+}
+
+func TestEventDispatcher_ResourceConfigFromReplies(t *testing.T) {
+	testCases := []struct {
+		isolators     []*lifecycle.IsolationControl
+		resources     *ResourceConfig
+		shouldBe      string
+		requestedKind lifecycle.IsolationControl_Kind
+	}{
+		{
+			isolators: []*lifecycle.IsolationControl{
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+					Value: "5",
+				},
+			},
+			resources:     &ResourceConfig{},
+			shouldBe:      "5",
+			requestedKind: lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+		},
+		{
+			isolators: []*lifecycle.IsolationControl{
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+					Value: "1",
+				},
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+					Value: "2",
+				},
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_MEMS,
+					Value: "3",
+				},
+			},
+			resources:     &ResourceConfig{},
+			shouldBe:      "2",
+			requestedKind: lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+		},
+		{
+			isolators: []*lifecycle.IsolationControl{
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_MEMS,
+					Value: "1",
+				},
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_CPUS,
+					Value: "2",
+				},
+				{
+					Kind:  lifecycle.IsolationControl_CGROUP_CPUSET_MEMS,
+					Value: "3",
+				},
+			},
+			resources:     &ResourceConfig{},
+			shouldBe:      "3",
+			requestedKind: lifecycle.IsolationControl_CGROUP_CPUSET_MEMS,
+		},
+	}
+
+	for _, testCase := range testCases {
+		er := &lifecycle.EventReply{
+			IsolationControls: testCase.isolators,
+		}
+		ed := newEventDispatcher()
+		output := ed.ResourceConfigFromReplies(er, testCase.resources)
+
+		switch testCase.requestedKind {
+		case lifecycle.IsolationControl_CGROUP_CPUSET_CPUS:
```

Comment:
to consider: instead doing `switch case` here, adjust the `testCases` to consist of output from isolators and expected `ResourceConfig`. the in the loop you just need to compare those. The advantage is that adding new test cases involves only appending to `testCase` structure.